tgui/packages/tgui/interfaces/ZoldorfPlayerShop/index.tsx

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Glowbold (https://github.com/pgmzeta)
+ * @license MIT
+ */
+
+import { InfernoNode } from 'inferno';
+import { useBackend } from '../../backend';
+import { Box, Button, Flex, Section, Stack } from '../../components';
+import { Window } from '../../layouts';
+import { isSoulProductData } from './type';
+import type { ZoldorfCommonProductData, ZoldorfPlayerShopData } from './type';
+
+export const ZoldorfPlayerShop = (_, context) => {
+  const { act, data } = useBackend<ZoldorfPlayerShopData>(context);
+  const { products, credits } = data;
+  return (
+    <Window width="500" height="600">
+      <Window.Content>
+        <Stack vertical fill>
+          <Stack.Item grow>
+            <Section fill scrollable>
+              <Stack vertical>
+                {products.map((product) => {
+                  return (
+                    <ZoldorfProductListItem {...product} key={product.name}>
+                      {isSoulProductData(product) ? (
+                        <Button
+                          color="red"
+                          content={`${product.soul_percentage}%`}
+                          disabled={product.soul_percentage > data.user_soul}
+                          onClick={() => act('soul_purchase', { 'item': product.name })}
+                          style={{
+                            'width': '50px',
+                          }}
+                        />
+                      ) : (
+                        <Button
+                          color="green"
+                          content={`${product.price}⪽`}
+                          disabled={product.price > credits}
+                          onClick={() => act('credit_purchase', { 'item': product.name })}
+                          width="50px"
+                        />
+                      )}
+                    </ZoldorfProductListItem>
+                  );
+                })}
+              </Stack>
+            </Section>
+          </Stack.Item>
+          {credits !== 0 && (
+            <Stack.Item bold>
+              <Box inline>{`Cash: ${credits}⪽`}</Box>
+              <Button ml="5px" icon="eject" content={'eject'} onClick={() => act('returncash')} />
+            </Stack.Item>
+          )}
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+interface ZoldorfProductListItemProps extends ZoldorfCommonProductData {
+  children: InfernoNode;
+}
+
+const ZoldorfProductListItem = (props: ZoldorfProductListItemProps) => {
+  const { name, img, stock, infinite, children } = props;
+  return (
+    <Flex
+      style={{
+        'align-items': 'center',
+        'border-bottom': '1px #555 solid',
+      }}>
```

Comment:
used a section as requested, but I feel like the current vending machines use a bottom border more as a left-to-right rule line since the purchase buttons are right-aligned.

Reply:
Fair I suppose. At this point we probably need to think about having "list of items with images and buttons" as a standard interface somewhere, but not needed for your PR.

I'm fine with you putting the border back in for the sake of UI consistency, apologies for back-and-forth on it.

Reply:
added with comments for future clarity